app/models/manageiq/providers/openstack/cloud_manager/host_aggregate.rb

```diff
@@ -1,8 +1,118 @@
 class ManageIQ::Providers::Openstack::CloudManager::HostAggregate < ::HostAggregate
+  include SupportsFeatureMixin
+
+  supports :update_aggregate
+  supports :delete_aggregate
+  supports :add_host
+  supports :remove_host
+
   store :metadata, :accessors => [:availability_zone]
 
   # if availability zone named in metadata exists, return it
   def availability_zone_obj
     AvailabilityZone.find_by_ems_ref_and_ems_id(availability_zone, ems_id)
   end
+
+  def self.create_aggregate(ext_management_system, options)
+    raise ArgumentError, _("ext_management_system cannot be nil") if ext_management_system.nil?
+
+    create_args = {:name => options[:name]}
+    if options[:availability_zone]
+      create_args[:availability_zone] = options[:availability_zone]
+    end
+    aggregate = nil
+    metadata = {}
+
+    connection_options = {:service => "Compute"}
+    ext_management_system.with_provider_connection(connection_options) do |service|
+      aggregate = service.aggregates.create(create_args)
+      if aggregate.availability_zone
+        metadata[:availability_zone] = aggregate.availability_zone
+      end
+    end
+    create(:name                  => options[:name],
+           :ems_ref               => aggregate.id,
+           :metadata              => metadata,
+           :ext_management_system => ext_management_system)
+  rescue => e
+    _log.error "host_aggregate=[#{options[:name]}], error: #{e}"
+    raise MiqException::MiqHostAggregateCreateError, e.to_s, e.backtrace
+  end
+
+  def external_aggregate
+    connection_options = { :service => "Compute" }
+    ext_management_system.with_provider_connection(connection_options) do |service|
+      service.aggregates.get(ems_ref)
+    end
+  end
+
+  def update_aggregate(options)
+    unless options[:name].blank?
+      rename_aggregate(options[:name])
+    end
+
+    if options[:metadata]
+      update_aggregate_metadata(options[:metadata])
+    end
+  end
+
+  def rename_aggregate(new_name)
```

Comment:
we tend to prefix methods, that do external provider changes with `raw_` but as this is in ovirt code only, I'll leave this up to you :)

not specific to this method, but others in here too